### PR TITLE
Create categories

### PR DESCRIPTION
**Pull request by Taraskin777**
Show categories on the main page

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1 +1,2 @@
-<app-header/>
\ No newline at end of file
+<app-header/>
+<app-categories></app-categories>
```
Comment (kulbaba) [inline src/app/app.component.html]: Choose one style for component  
I think better 
<app-header></header>
<app-categories></app-categories>

File: src/app/core/services/data.service.ts [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+  price: number;
+}
+
+export interface Data {
+  id: number;
+  name: string;
+  image: string;
+  dishes: Dishes[];
+}
+
```
Comment (kulbaba) [inline src/app/core/services/data.service.ts]: "Data" is not a good name for an interface. It is difficult to understand what it is.

File: src/app/core/services/data.service.ts [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class DataService {
+  constructor(private http: HttpClient) {}
 
-  constructor() { }
+  public getData(): Observable<Data[]> {
+    return this.http.get<Data[]>('http://localhost:3000/categories');
+  }
```
Comment (kulbaba) [inline src/app/core/services/data.service.ts:23]: 'url' for endpoint is better to take it to the environment. 
 
 
 'url' = 'http://localhost:3000'

File: src/app/menu/components/categories/categories.component.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  styleUrls: ['./categories.component.css'],
+})
+export class CategoriesComponent implements OnInit {
+  categories: Data[] = [];
+
+  constructor(private dataService: DataService) {}
+
+  ngOnInit(): void {
+    this.dataService.getData().subscribe(
+      (data: Data[]) => {
```
Comment (kulbaba) [inline src/app/menu/components/categories/categories.component.ts]: add unsubscribe

**Review by kulbaba — CHANGES_REQUESTED** [review]

File: src/app/app.component.html (hunk 1 above)
Comment (Taraskin777) [reply src/app/app.component.html]: +

File: src/app/core/services/data.service.ts (hunk 2 above)
Comment (Taraskin777) [reply src/app/core/services/data.service.ts]: Done

File: src/app/core/services/data.service.ts (hunk 3 above)
Comment (Taraskin777) [reply src/app/core/services/data.service.ts:23]: done

**Pull request merged**